src/util/sharing_map.h

```diff
@@ -1001,13 +1098,138 @@ SHARING_MAPT(void)::erase(const key_type &k)
   num--;
 }
 
+SHARING_MAPT2(, innert *)::migrate(
+  const std::size_t step,
+  const std::size_t key_suffix,
+  const std::size_t bit_last,
+  innert *ip)
+{
+  SM_ASSERT(step < steps - 1);
+  SM_ASSERT(ip != nullptr);
+  SM_ASSERT(ip->is_defined_internal());
+
+  const innert &child = *ip->find_child(bit_last);
+  SM_ASSERT(child.is_defined_container());
+
+  const leaf_listt &ll = child.get_container();
+
+  // Only containers at the bottom can contain more than two elements
+  SM_ASSERT(is_singular(ll));
+
+  const leaft &leaf = ll.front();
+  std::size_t key_existing = hash()(leaf.get_key());
+
+  key_existing >>= chunk * step;
+
+  // Copy the container
+  innert container_copy(child);
+
+  // Delete existing container
+  ip->remove_child(bit_last);
+
+  // Add internal node
+  ip = ip->add_child(bit_last);
+  SM_ASSERT(ip->empty());
+
+  // Find place for both elements
+
+  std::size_t i = step + 1;
+  std::size_t key = key_suffix;
+
+  key_existing >>= chunk;
+  key >>= chunk;
+
+  SM_ASSERT(i < steps);
+
+  do
+  {
+    std::size_t bit_existing = key_existing & mask;
+    std::size_t bit = key & mask;
+
+    if(bit != bit_existing)
+    {
+      // Place found
+
+      innert *cp2 = ip->add_child(bit_existing);
+      cp2->swap(container_copy);
+
+      innert *cp1 = ip->add_child(bit);
+      return cp1;
+    }
+
+    SM_ASSERT(bit == bit_existing);
+    ip = ip->add_child(bit);
+
+    key >>= chunk;
+    key_existing >>= chunk;
+
+    i++;
+  } while(i < steps);
+
+  leaft leaf_copy(as_const(&container_copy)->get_container().front());
+  leaf_listt &c = ip->get_container();
+  c.push_front(leaf_copy);
```

Comment:
`c` doesn't seem to add much value: `ip->get_container().push_front(leaf_copy);`